lib/concurrent-ruby/concurrent/atomic/mutex_semaphore.rb

```diff
@@ -48,13 +57,23 @@ def try_acquire(permits = 1, timeout = nil)
       Utility::NativeInteger.ensure_integer_and_bounds permits
       Utility::NativeInteger.ensure_positive permits
 
-      synchronize do
+      acquired = synchronize do
         if timeout.nil?
           try_acquire_now(permits)
         else
           try_acquire_timed(permits, timeout)
         end
       end
+
+      if acquired && block_given?
+        begin
+          yield
+        ensure
+          release(permits)
+        end
+      end
```

Comment:
As I was inlining this I wondered about the API here.

Given this is `TRY_acquire`, which is correct:

- `yield` if acquired
- always `yield(acquired)` and let the block decide what to do

My gut is the **former**, as it is more intuitive, and less likely someone would forget to check if the lock was acquired or not, but I thought I should check, as there could be cases where it might be useful to do something in the block if the lock failed to be acquired. Though maybe that use case should just use the non-block version... 🤷 

If we did want the latter, we could do something like

```suggestion
      if block_given?
        begin
          yield(acquired)
        ensure
          release(permits) if acquired
        end
      end
```